rust_snuba/src/runtime_config.rs

```diff
@@ -40,12 +40,56 @@ pub fn get_str_config(key: &str) -> Result<Option<String>, Error> {
 
 #[cfg(test)]
 mod tests {
+
     use super::*;
 
     #[test]
     fn test_runtime_config() {
         crate::testutils::initialize_python();
-        let config = get_str_config("test");
+        let config = get_str_config("key0");
+        assert_eq!(config.unwrap(), None);
+        CONFIG.write().clear();
+    }
+
+    #[test]
+    fn test_runtime_config_is_stored_in_cache() {
+        crate::testutils::initialize_python();
+        assert!(!CONFIG.read().contains_key("key1"));
+        let config = get_str_config("key1");
+        assert_eq!(config.unwrap(), None);
+        assert!(CONFIG.read().contains_key("key1"));
+        CONFIG.write().clear();
+    }
+
+    #[test]
+    fn test_runtime_config_retrieves_from_cache_within_deadline() {
+        {
+            CONFIG.write().insert(
+                "key2".to_string(),
+                (
+                    Some("value2".to_string()),
+                    Deadline::new(Duration::from_secs(10)),
+                ),
+            );
+        }
+        let config = get_str_config("key2");
+        assert_eq!(config.unwrap(), Some("value2".to_string()));
+        CONFIG.write().clear();
+    }
+
+    #[test]
+    fn test_runtime_config_invalidates_cache_outside_deadline() {
+        {
+            CONFIG.write().insert(
+                "key3".to_string(),
+                (
+                    Some("value3".to_string()),
+                    Deadline::new(Duration::from_secs(0)),
+                ),
+            );
+        }
+        let config = get_str_config("key3");
         assert_eq!(config.unwrap(), None);
+        CONFIG.write().clear();
```

Comment:
I wanted to set this deadline to maybe 1 sec from now, and sleep for 1 sec to retrieve it, like so
```rust
#[test]
    fn test_runtime_config_invalidates_cache_outside_deadline() {
        {
            CONFIG.write().insert(
                "key3".to_string(),
                (
                    Some("value3".to_string()),
                    Deadline::new(Duration::from_secs(1)),
                ),
            );
        }
        thread::sleep(Duration::from_secs(900));
        let config = get_str_config("key3");
        assert_eq!(config.unwrap(), None);
        CONFIG.write().clear();
    }
```
But for some reason this fails, and the debugger shows that `coarsetime::Instant::recent()` does not appear to be advancing

Reply:
these tests will run concurrently -- the easiest way to get around this issue is to run all tests in one function.